src/software/ai/hl/stp/action/intercept_ball_action.cpp

```diff
@@ -35,146 +36,132 @@ void InterceptBallAction::accept(MutableActionVisitor& visitor)
 {
-    Ray ball_ray(ball.position(), ball.velocity());
-    if (field.pointInFieldLines(ball.position()))
+    // Run into the ball until the ball is in the dribbler and the ball's velocity
+    // is similar to the robot's (so we know we have control of the ball)
+    bool intercept_done = false;
+    do
     {
-        std::unordered_set<Point> intersections =
-            intersection(field.fieldLines(), ball_ray);
-        if (!intersections.empty())
+        auto face_ball_orientation = (ball.position() - robot->position()).orientation();
+        yield(std::make_unique<MoveIntent>(
+            robot->id(), ball.position(), face_ball_orientation, 0, 0, DribblerEnable::ON,
+            MoveType::NORMAL, AutokickType::NONE, BallCollisionType::ALLOW));
+
+        if (ball.velocity().length() > BALL_MOVING_SLOW_SPEED_THRESHOLD * 2)
         {
-            return *intersections.begin();
+            restart();
         }
-    }
 
-    return std::nullopt;
-}
-
-void InterceptBallAction::calculateNextIntent(IntentCoroutine::push_type& yield)
-{
-    // This action tries to intercept and collect the ball on the field.
-    // We find the point on the Ray formed by the ball's velocity that is closest to
-    // the robot. If the time for the robot to get to that point is less than the time
-    // for the ball to get to that point, the robot moves to that point to intercept the
-    // ball. Once the robot is in the ball's path, it will move to a point just in
-    // front of the ball in order to meet it quicker, while staying in it's path.
-    //
-    // If the robot cannot reach the closest point on the ray before the ball, the robot
-    // will move to intercept the ball at the point it would leave the field. This
-    // generally causes the robot to move far enough ahead of the ball that it can switch
-    // to intercepting at the closest point on the ray like above.
-    //
-    // Finally, if the ball is moving slowly the robot will go directly to the ball.
+        bool ball_very_close_to_robot_dribbler =
+            distance(ball.position(),
+                     robot->position() + Vector::createFromAngle(robot->orientation())
+                                             .normalize(DIST_TO_FRONT_OF_ROBOT_METERS)) <
+            0.2 + BALL_MAX_RADIUS_METERS;
+        Vector dribbler_tangential_velocity =
+            Vector::createFromAngle(robot->orientation())
+                .perpendicular()
+                .normalize((DIST_TO_FRONT_OF_ROBOT_METERS + BALL_MAX_RADIUS_METERS) *
+                           robot->angularVelocity().toRadians());
+        Vector expected_ball_velocity = robot->velocity() + dribbler_tangential_velocity;
+        bool velocity_magnitudes_similar =
+            std::fabs(expected_ball_velocity.length() - ball.velocity().length()) < 0.05;
+        bool velocity_angles_similar =
+            expected_ball_velocity.orientation().minDiff(ball.velocity().orientation()) <
+            Angle::fromDegrees(10);
```

Comment:
this should be a constant if not a dyanamic param

Reply:
made constant